packages/calcite-components/src/components/tab-nav/tab-nav.tsx

```diff
@@ -241,13 +240,17 @@ export class TabNav extends LitElement {
 
   private internalActivateTabHandler(event: CustomEvent<TabChangeEventDetail>): void {
     const activatedTabTitle = event.target as TabTitle["el"];
+    const currentSelectedTabTitle = this.selectedTitle;
 
     this.selectedTabId = event.detail.tab
       ? event.detail.tab
       : this.getIndexOfTabTitle(activatedTabTitle);
     event.stopPropagation();
 
     this.selectedTitle = activatedTabTitle;
+    if (currentSelectedTabTitle !== activatedTabTitle && event.detail.userTriggered) {
```

Comment:
nitpick: Highly unlikely, but what if two tabs have the same title or no title at all? Is there a better unique identifier we could use here? Maybe an undocumented guid property?

Reply:
Great point, updated to use `id`.

Reply:
@anveshmekala what if the tab doesn't have an id? Id isn't set automatically right?

Reply:
Oh, it is. nevermind :)